Gems/Camera/Code/Source/CameraComponentController.cpp

```diff
@@ -139,13 +139,7 @@ namespace Camera
 
     void CameraComponentController::DeactivateAtomView()
     {
-        if (!IsActiveView())
```

Comment:
Why remove this?

Reply:
Sorry I meant to add a comment here on the review (I forgot as I moved this to draft for now as it needs more time)...

So this was actually leading to a crash in the `onChange` callback in the `Init` function of `CameraComponentController`. `onChange` captures `CameraComponentController` by value and assigns it to `m_atomCameraViewGroup`, which is a `shared_ptr`. Ownership of this is then shared by passing it to `PushViewGroup` (which stores the `shared_ptr`). When `CameraComponentController` gets destroyed and doesn't remove the reference to `m_atomCameraViewGroup`, it has a dangling pointer to the `CameraComponentController` instance which no longer exists, so we have to make sure to also remove it.

There might be a different way to resolve it but this felt like the most correct fix to me.

Reply:
Note: This crash only happens when you have more than one camera component in the scene and have one of the camera selected when moving to/from Game Mode